src/vs/editor/contrib/codelens/codelensController.ts

```diff
@@ -402,6 +405,66 @@ export class CodeLensContribution implements IEditorContribution {
 			}
 		});
 	}
+
+	public getLenses(): CodeLensWidget[] {
+		return this._lenses;
+	}
```

Comment:
Why expose the widgets and not the model itself?

Reply:
We need the line number of the widget on line 421:

```const activeLensesWidgets = codelensController.getLenses().filter(lens => lens.getLineNumber() === lineNumber);```

Reply:
Ah, the live/updated line numbers... That makes good sense. Thanks for clarifying